### PR TITLE
[MODEL-9648] ViT epochs as hyperparameter

### PR DESCRIPTION
**Pull request by scottp-ml**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
This adds epochs as a tunable hyperparameter for the ViT template.  Also I removed some unused imports from other templates.

### CODE REVIEW

File: task_templates/2_estimators/11_python_huggingface_vit/custom.py [hunk 1]
```diff
@@ -30,7 +30,13 @@ def __len__(self):
 
 
 class CustomTask(BinaryEstimatorInterface):
-    def fit(self, X, y, row_weights=None, **kwargs):
+    def fit(self, X, y, row_weights=None, parameters=None, **kwargs):
```
Comment (yakov-g) [inline task_templates/2_estimators/11_python_huggingface_vit/custom.py:33]: If you want `parameters` to be required param, maybe make it positional t.e required?!

Comment (yakov-g) [reply task_templates/2_estimators/11_python_huggingface_vit/custom.py:33]: Actually,
maybe just pass `epochs` directly as `kwargs` and check it is not None. i.e.
```
epochs = kwargs.get("epochs")
if epochs is None:
   raise ValueError("epoch has to be provided in kwargs and can not be None")
```

Comment (scottp-ml) [reply task_templates/2_estimators/11_python_huggingface_vit/custom.py:33]: I believe all hyperparameters are added to parameters, they are not accessible individually.  This is passed by drum from reading the model-metadata

Comment (yakov-g) [reply task_templates/2_estimators/11_python_huggingface_vit/custom.py:33]: oh, I see, 
ok then

**Review by yakov-g — APPROVED** [review]

**Review by gkoundry — APPROVED** [review]
LGTM Core Modeling

**Pull request merged**